src/main_thread/init/utils/initial_seek_and_play.ts

```diff
@@ -107,18 +108,28 @@ export default function performInitialSeekAndPlay(
       if (!isDirectfile || typeof startTime === "number") {
         const initiallySeekedTime =
           typeof startTime === "number" ? startTime : startTime();
-        if (initiallySeekedTime !== 0) {
+        if (initiallySeekedTime !== 0 && initiallySeekedTime !== undefined) {
           performInitialSeek(initiallySeekedTime);
         }
         waitForSeekable();
       } else {
         playbackObserver.listen(
           (obs, stopListening) => {
-            if (obs.readyState >= 1) {
+            const initiallySeekedTime =
+              typeof startTime === "number" ? startTime : startTime();
+            const isSafari = isSafariDesktop || isSafariMobile;
+
+            // For most browsers, readyState 1 is sufficient for seeking.
+            // However, on Safari, readyState 1 may occur too early, leading to cases where
+            // mediaElement.duration or mediaElement.seekable.end(0) are not yet available
+            // and could return `Infinity`, causing issues with seeking.
+            // To handle this, we require readyState 2 on Safari before attempting to seek.
+            const minReadyStateForSeeking =
+              isSafari && initiallySeekedTime === undefined ? 2 : 1;
```

Comment:
We could isolate that to `compat` to have all device work-arounds visible in the same directory.

Reply:
Done